src/pages/Balance/partials/BalanceBody.tsx

```diff
@@ -0,0 +1,105 @@
+import { CircularProgress } from "@mui/material";
+import { useBalanceContext } from "../context";
+import AppearFadeIn from "@components/Animation/AppearFadeIn";
+import { useEffect, useState } from "react";
+import AppearGrow from "@components/Animation/AppearGrow";
+import { cn } from "@utils/cn";
+import glassmorphism from "@utils/glassmorphism";
+
+const BalanceBody = () => {
+  const { state } = useBalanceContext();
+  const [firstNewDate, setFirstNewDate] = useState(true);
+  useEffect(() => {
+    if (firstNewDate) {
+      setFirstNewDate(false);
+    }
+  }, [firstNewDate, setFirstNewDate]);
+
+  return (
+    <AppearFadeIn direction="bottom" delay={0.8}>
+      <div
+        className={cn(
+          "flex flex-col gap-4 p-4 rounded-md",
+          glassmorphism({ container: true, border: true }),
+        )}
+      >
+        <div className="text-lg font-semibold">Latest 10 Balance History</div>
+        <hr />
+        {state.balanceHistoryLoading ? (
+          <CircularProgress size="2rem" />
+        ) : (
+          <div
+            className={cn(
+              "overflow-x-auto rounded-md",
+              glassmorphism({ container: true, border: true }),
+            )}
+          >
+            {state.balanceHistory.length === 0 ? (
+              <div className="text-center p-2">Balance History is empty</div>
+            ) : (
+              <AppearGrow trigger direction="x">
+                {state.balanceHistory.map((item, index) => {
+                  return (
+                    <AppearFadeIn
+                      key={index}
+                      direction="bottom"
+                      delay={0.1 * index}
+                    >
+                      <div
+                        className={cn(
+                          "p-4 flex justify-between group",
+                          glassmorphism({ hover: true }),
+                        )}
+                      >
+                        <div className="flex flex-col gap-1 font-medium group-hover:text-gray-300">
+                          <div className="flex items-center gap-4">
```

Comment:
flex flex-col tapi children cuma 1, jadi unecessary jg, usahain jgn terlalu banyak children
![image](https://github.com/UKM-Programming-UMDP/procom-kas-fe-user/assets/91197642/ae722638-db36-4c79-b695-7723672966ef)

Reply:
atau untuk yang component when result not loading & not empty, pindahin ke component terpisah: BalanceBodyContent